### PR TITLE
[new release] re (1.12.0)

### PR DESCRIPTION
**Pull request by rgrinberg**
RE is a regular expression library for OCaml

- Project page: <a href="https://github.com/ocaml/ocaml-re">https://github.com/ocaml/ocaml-re</a>

##### CHANGES:

* Add `Re.split_delim` (ocaml/ocaml-re#233)
* Fix handling of empty matches in splitting and substitution functions (ocaml/ocaml-re#233)
* Add support for character classes in `Re.Posix` (ocaml/ocaml-re#263)

### CODE REVIEW

**Comment by shonfeder** [comment]
Thanks for the update :)

Looks like we have a variety of relevant CI errors:

ounit missing as a dependency:

```
- Error: Library "ounit2" not found.
- -> required by _build/default/lib_test/test_str.exe
- -> required by alias lib_test/runtest in lib_test/dune:10
[ERROR] The compilation of re.1.12.0 failed at "dune runtest -p re -j 31".

#=== ERROR while compiling re.1.12.0 ==========================================#
# context              2.3.0~alpha~dev | linux/x86_64 | ocaml-base-compiler.4.12.1 | pinned(https://github.com/ocaml/ocaml-re/releases/download/1.12.0/re-1.12.0.tbz)
# path                 ~/.opam/4.12/.opam-switch/build/re.1.12.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune runtest -p re -j 31
# exit-code            1
# env-file             ~/.opam/log/re-7-1aa967.env
# output-file          ~/.opam/log/re-7-1aa967.out
### output ###
# File "lib_test/dune", line 2, characters 25-31:
# 2 |  (libraries re fort_unit ounit2)
#                              ^^^^^^
# Error: Library "ounit2" not found.
# -> required by _build/default/lib_test/test_easy.exe
# -> required by alias lib_test/runtest in lib_test/dune:4
# File "lib_test/dune", line 8, characters 29-35:
# 8 |  (libraries re fort_unit str ounit2)
#                                  ^^^^^^
# Error: Library "ounit2" not found.
# -> required by _build/default/lib_test/test_str.exe
# -> required by alias lib_test/runtest in lib_test/dune:10



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+- The following actions failed
| - build re 1.12.0
```

Missing values for ocaml 4.11 to 4.08:

```
[ERROR] The compilation of re.1.12.0 failed at "dune build -p re -j 31".

#=== ERROR while compiling re.1.12.0 ==========================================#
# context              2.3.0~alpha~dev | linux/x86_64 | ocaml-base-compiler.4.11.2 | pinned(https://github.com/ocaml/ocaml-re/releases/download/1.12.0/re-1.12.0.tbz)
# path                 ~/.opam/4.11/.opam-switch/build/re.1.12.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p re -j 31
# exit-code            1
# env-file             ~/.opam/log/re-7-4c02fa.env
# output-file          ~/.opam/log/re-7-4c02fa.out
### output ###
# (cd _build/default && /home/opam/.opam/4.11/bin/ocamlc.opt -w -40 -w -50 -g -bin-annot -I lib/.re.objs/byte -I /home/opam/.opam/4.11/lib/seq -intf-suffix .ml -no-alias-deps -open Re__ -o lib/.re.objs/byte/re__Cset.cmo -c -impl lib/cset.ml)
# File "lib/cset.ml", line 35, characters 12-22:
# 35 | let equal = List.equal ~eq:(fun (x, y) (x', y') -> Int.equal x x' && Int.equal y y')
#                  ^^^^^^^^^^
# Error: Unbound value List.equal
# (cd _build/default && /home/opam/.opam/4.11/bin/ocamlc.opt -w -40 -w -50 -g -bin-annot -I lib/.re.objs/byte -I /home/opam/.opam/4.11/lib/seq -intf-suffix .ml -no-alias-deps -open Re__ -o lib/.re.objs/byte/re__Automata.cmo -c -impl lib/automata.ml)
# File "lib/automata.ml", line 105, characters 4-14:
# 105 |     List.equal ~eq:(fun (x, y) (x', y') -> Int.equal x x' && Int.equal y y') marks t.marks
#           ^^^^^^^^^^
# Error: Unbound value List.equal
# (cd _build/default && /home/opam/.opam/4.11/bin/ocamlopt.opt -w -40 -w -50 -g -I lib/.re.objs/byte -I lib/.re.objs/native -I /home/opam/.opam/4.11/lib/seq -intf-suffix .ml -no-alias-deps -open Re__ -o lib/.re.objs/native/re__Cset.cmx -c -impl lib/cset.ml)
# File "lib/cset.ml", line 35, characters 12-22:
# 35 | let equal = List.equal ~eq:(fun (x, y) (x', y') -> Int.equal x x' && Int.equal y y')
#                  ^^^^^^^^^^
# Error: Unbound value List.equal
# (cd _build/default && /home/opam/.opam/4.11/bin/ocamlc.opt -w -40 -w -50 -g -bin-annot -I lib/.re.objs/byte -I /home/opam/.opam/4.11/lib/seq -intf-suffix .ml -no-alias-deps -open Re__ -o lib/.re.objs/byte/re__Ast.cmo -c -impl lib/ast.ml)
# File "lib/ast.ml", line 12, characters 36-46:
# 12 |   | Alternative a, Alternative b -> List.equal ~eq a b
#                                          ^^^^^^^^^^
# Error: Unbound value List.equal
```

Syntax errors and missing modules for ocaml =< 4.08

```
[ERROR] The compilation of re.1.12.0 failed at "dune build -p re -j 39".

#=== ERROR while compiling re.1.12.0 ==========================================#
# context              2.3.0~alpha~dev | linux/x86_64 | ocaml-base-compiler.4.07.1 | pinned(https://github.com/ocaml/ocaml-re/releases/download/1.12.0/re-1.12.0.tbz)
# path                 ~/.opam/4.07/.opam-switch/build/re.1.12.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p re -j 39
# exit-code            1
# env-file             ~/.opam/log/re-7-b2bb93.env
# output-file          ~/.opam/log/re-7-b2bb93.out
### output ###
# (cd _build/default && /home/opam/.opam/4.07/bin/ocamldep.opt -modules -intf lib/fmt.mli) > _build/default/lib/.re.objs/re__Fmt.intf.d
# File "lib/fmt.mli", line 1, characters 15-17:
# Error: Syntax error
# (cd _build/default && /home/opam/.opam/4.07/bin/ocamlc.opt -w -40 -w -50 -g -bin-annot -I lib/.re.objs/byte -I /home/opam/.opam/4.07/lib/seq -no-alias-deps -open Re__ -o lib/.re.objs/byte/re__Import.cmo -c -impl lib/import.ml)
# File "lib/import.ml", line 7, characters 12-21:
# Error: Unbound module Int
```

**Comment by rgrinberg** [comment]
Thanks. I suppose we can just bump the minimum version of OCaml and disable the tests for this release.

**Comment by rgrinberg** [comment]
Ounit is listed as a dependency:

> "ounit" {with-test}

Maybe there's a missing lower bound?

File: packages/re/re.1.12.0/opam [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+dev-repo: "git+https://github.com/ocaml/ocaml-re.git"
+
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+]
+
+depends: [
+  "ocaml" {>= "4.03"}
```
Comment (mseri) [inline packages/re/re.1.12.0/opam:23]:
```suggestion
  "ocaml" {>= "4.12"}
```

File: packages/re/re.1.12.0/opam [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  "dune" {>= "2.0"}
+  "ounit" {with-test}
```
Comment (mseri) [inline packages/re/re.1.12.0/opam:25]:
```suggestion
  "ounit2" {with-test}
```
It was introduced with ounit 2.2.0, which simply installs the ounit2 package. I think this should be enough to fix the error

**Review by shonfeder — APPROVED** [review]

**Comment by shonfeder** [comment]
Only revdeps are failing here, and none of them appear to be due to this update. All else looks good to me. Thanks for publishing the update!

**Pull request merged**